include/seqan3/alignment/pairwise/detail/policy_affine_gap_recursion.hpp

```diff
@@ -201,5 +201,21 @@ class policy_affine_gap_recursion
                 new_horizontal_score,
                 previous_cell.horizontal_score() + gap_open_score};
     }
+
+    /*!\brief Returns the lowest viable score.
+     *
+     * \details
+     *
+     * In some versions of the algorithms a value representing minus infinity is needed. Since the data type is an
+     * signed integral there is no infinity but only the lowest possible value that can be represented by the score
+     * type. In order to avoid unnecessary if conditions to protect against signed integer underflow the lowest viable
+     * score is computed. Subtracting a gap penalty from this will still result in a valid score which represents
+     * minus infinity.
+     */
+    score_type lowest_viable_score() const noexcept
+    {
+        assert(m_gap_open_score <= 0 && m_gap_extension_score <= 0);
+        return std::numeric_limits<score_type>::lowest() - (m_gap_open_score + m_gap_extension_score);
```

Comment:
That can only happen if the score would overflow/underflow in general because the score range is too small. But in this case we simply say that there is undefined behaviour. What we can say is, however, what would be the minimal viable score such that we do not underflow when subtracting a value from this. This has two effects: In the banded case I spare one additional case to consider when computing the score of the last cell of the banded column. Later in the local vectorised score I can choose this score as the new `0`, such that I can use as much of the signed score range as possible without using saturated integer arithmetic which is not supported on all vector extensions.